linode/manager
# fix: [M3-8970] - Duplicate VPCs fetch on Linode Create

## Description 📝

- Multiple fetches to were being made to the VPCs endpoint on the Linode Create flow. Only one should be necessary.
- This PR reactors things a bit to resolve the extra fetching 🔄 
- Also fixes unreported bug in which the user would only see their first 100 VPCs in the selected region
  - Fixed by using `useAllVPCsQuery` instead of paginated `useVPCsQuery`
  - In the future we can implement infinite scroll if we are worried about users having thousands of VPCs

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/user-attachments/assets/eeece4f2-58e9-48e7-b915-adc6acc33dd1" /> | <video src="https://github.com/user-attachments/assets/10d18f42-5a56-49ea-a105-a2589012160e" />  |


## How to test 🧪

- Go to Linode Create with the network tab open 🌐 
- Observe fetches to the VPC endpoint while you load and interact with the Linode Create flow
- Verify VPCs only fetch once (will re-fetch if you change regions)

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>
